src/hmem_cuda.c

```diff
@@ -463,51 +463,49 @@ static void cuda_hmem_dl_cleanup(void)
 static int cuda_hmem_verify_devices(void)
 {
 	nvmlReturn_t nvml_ret;
+        cudaError_t cuda_ret;
 	unsigned int nvml_device_count = 0;
-	
+
 	/* Check w/ nvmlDeviceGetCount_v2() first, to avoid more expensive
 	 * call to cudaGetDeviceCount() when possible.
 	 */
 
 	/* Verify NVIDIA devices are present on the host. */
 	nvml_ret = ofi_nvmlDeviceGetCount_v2(&nvml_device_count);
-	if (NVML_SUCCESS == nvml_ret) {
-		FI_INFO(&core_prov, FI_LOG_CORE,
-			"Number of NVIDIA devices detected: %u\n",
-			nvml_device_count);
-
-		if (nvml_device_count > 0) {
-			cudaError_t cuda_ret;
-
-			/* Verify CUDA compute-capable devices are present on the host. */
-			cuda_ret = ofi_cudaGetDeviceCount(&cuda_attr.device_count);
-			switch (cuda_ret) {
-			case cudaSuccess:
-				break;
-
-			case cudaErrorNoDevice:
-				return -FI_ENOSYS;
-
-			default:
-				FI_WARN(&core_prov, FI_LOG_CORE,
-					"Failed to perform cudaGetDeviceCount: %s:%s\n",
-					ofi_cudaGetErrorName(cuda_ret),
-					ofi_cudaGetErrorString(cuda_ret));
-				return -FI_EIO;
-			}
-
-			FI_INFO(&core_prov, FI_LOG_CORE,
-				"Number of CUDA devices detected: %d\n",
-				cuda_attr.device_count);
-		} else {
-			cuda_attr.device_count = 0;
-		}
-
-		if (cuda_attr.device_count <= 0)
-			return -FI_ENOSYS;
-	} else {
-		return -FI_ENOSYS;
-	}
+	if (nvml_ret != NVML_SUCCESS)
+                return -FI_ENOSYS;
+
+        FI_INFO(&core_prov, FI_LOG_CORE,
+                "Number of NVIDIA devices detected: %u\n",
+                nvml_device_count);
+
+        if (nvml_device_count > 0) {
+                /* Verify CUDA compute-capable devices are present on the host. */
+                cuda_ret = ofi_cudaGetDeviceCount(&cuda_attr.device_count);
+                switch (cuda_ret) {
+                case cudaSuccess:
+                        break;
+
+                case cudaErrorNoDevice:
+                        return -FI_ENOSYS;
+
+                default:
+                        FI_WARN(&core_prov, FI_LOG_CORE,
+                                "Failed to perform cudaGetDeviceCount: %s:%s\n",
+                                ofi_cudaGetErrorName(cuda_ret),
+                                ofi_cudaGetErrorString(cuda_ret));
+                        return -FI_EIO;
+                }
+
+                FI_INFO(&core_prov, FI_LOG_CORE,
+                        "Number of CUDA devices detected: %d\n",
+                        cuda_attr.device_count);
+        } else {
+                cuda_attr.device_count = 0;
+        }
+
+        if (cuda_attr.device_count <= 0)
+                return -FI_ENOSYS;
```

Comment:
Don't do updates in a separate patch.  Just update the original patch directly.  Otherwise, there are potentially 2 patches that need to be picked into the stable branches.

For the original patch, please format the commit messages, so that the 1st line is only about 72 characters long.  It needs to be easily viewable using git log --oneline for fast identification and cherry picking of patches.  Add more details into the main message.  This seems like a very minor optimization for a function that's called once.  Some details on why we're adding a second get device count call into this path should be captured in the commit message.

Reply:
> Don't do updates in a separate patch. Just update the original patch directly. Otherwise, there are potentially 2 patches that need to be picked into the stable branches.
> 

Why not squash during the merge from my branch into main?

Reply:
>  This seems like a very minor optimization for a function that's called once. Some details on why we're adding a second get device count call into this path should be captured in the commit message.

nvmlDeviceGetCount is much lighter-weight than cudaGetDeviceCount and allows an early-out from the routine, improving performance for a simple simple IMB Allreduce test on AWS instances without GPUs by 40-300%.

Reply:
>Why not squash during the merge from my branch into main?

I don't quite follow. It is a convention in PRs that updates on a commit should be combined with that original commits via `git commit --amend`. There is no auto squash for github merge AFAIK

Reply:
> > Why not squash during the merge from my branch into main?
> 
> I don't quite follow. It is a convention in PRs that updates on a commit should be combined with that original commits via `git commit --amend`. There is no such kind of "auto squash" for github merge AFAIK

All of the projects i've contributed to have done this in different ways.  ¯\_(ツ)_/¯